keps/sig-network/4962-network-topology-standard/README.md

```diff
@@ -0,0 +1,867 @@
+- Define or implement network topology discovery mechanisms for CSPs or on-premises environments
+
+## Proposal
+
+We propose new node label type to capture network topology information:
+
+### Network Topology Label
+Format: `network.topology.kubernetes.io/<hierarchy-layer-type>: <name>`, where
+- `<hierarchy-layer-type>` defines the topology and characteristics of a network switch. The term `switch` may refer to a physical network device or a collection of closely connected devices with similar functionality.
+- `<name>` is a unique identifier for the hierarchy layer type.
+
+We propose to use the following four network hierarchy layer types:
+1. `accelerator`: Network interconnect for direct accelerator communication (e.g., Multi-node NVLink interconnect between NVIDIA GPUs)
+2. `block`: Rack-level switches connecting hosts in one or more racks as a block.
+3. `spine`: Spine-level switches connecting multiple blocks inside a datacenter.
+4. `datacenter`: Zonal switches connecting multiple datacenters inside an availability zone.
+
+These types will accommodate the majority of common network hierarchies across different CSP and on-prem environments.
+Having these labels available in Kubernetes clusters will help in designing cloud agnostic scheduling systems.
+The scheduler will prioritize hierarchy layers according to the order outlined above, providing a standardized
+approach for network-aware scheduling across a range of configurations.
+
+### User Stories
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Story 1
+
+As a data scientist running a data-intensive large-scale AI training job, I want to optimize the runtime
+by binding pods to nodes that are in close network proximity.
+This ensures better performance for my distributed workloads.
+
+Additionally, I do not want to modify the job specification when migrating the job across different Kubernetes environments.
+
+To achieve this, I can leverage the pod affinity feature of the default Kubernetes scheduler with topology keys:
+```yaml
+  spec:
+    affinity:
+      podAffinity:
+        preferredDuringSchedulingIgnoredDuringExecution:
+          - weight: 70
+            podAffinityTerm:
+              labelSelector:
+                matchExpressions:
+                  - key: app
+                    operator: In
+                    values:
+                      - training
+              topologyKey: network.topology.kubernetes.io/block
+          - weight: 90
+            podAffinityTerm:
+              labelSelector:
+                matchExpressions:
+                  - key: app
+                    operator: In
+                    values:
+                      - training
+              topologyKey: network.topology.kubernetes.io/accelerator
```

Comment:
Does this proposal address this user story? If so, can you add an example:

> As a data scientist, I want to ensure the pods are always placed within the same nvlink domain first before choosing another nvlink domain and I do not want to modify the job specification when migrating the job across different Kubernetes environments.

Reply:
Right, the `network.topology.kubernetes.io/accelerator` label value is unique per nvlink domain. But the user only provides the label name, i.e. `network.topology.kubernetes.io/accelerator`.
So, the scheduler will try to find a set of nodes with the same label value, hence placing the jobs in the same nvlink domain.
There is no need to modify  the job spec when migrating environments.
The User Story 1 describes this exact scenario.

Reply:
NVM, I mixed up the labelSector used in the example with the topologyKey, which sounds like when the same topologyKey is used, it is assumed that the scheduler will ensure pods are placed on the nodes with the require resources AND the same topologyKey value.